### PR TITLE
GAS-60: Alerting automation 

### PR DESCRIPTION
**Pull request by theTibi**
* Adding all Alert templates from current automation to new
* Adding filters controled by pmm_alert_filters
* Adding alerts only for specific services controled by pmm_alert_filters
* Removing percona specific variable names

### CODE REVIEW

File: roles/pmm/defaults/main.yaml [hunk 1]
```diff
@@ -22,9 +22,28 @@ pmm_agent_setup_flags:
...
+pmm_alertmanager_contactpoint_uid: '{{ pmm_alertamanger_default_contactpoint_uid | default("DefaultContact_")}}'
+pmm_templates_prefix: '{{ pmm_default_templates_prefix | default("DefaultPrefix_")}}'
+pmm_alerting_runbook_path: 'https://localhost/runbooks/'              # path used in alert labels
+pmm_alertmanager_rule_periods:                                                                          # Used to set the "for" duration in rules
+  interval_unit: m
+  low_interval: 1
+  med_interval: 5
+  high_interval: 10
+  max_interval: 15
+pmm_alert_filters: []            # example: '{"VmalertAlertsError":{"filters":[{"label":"env","regexp":"staging","type":"MATCH"},{"label":"node_name","regexp":"db10","type":"MISMATCH"}]}}'
```
Comment (cezmunsta) [inline roles/pmm/defaults/main.yaml]: @theTibi you have the variable defined as a list and then show a string. See [`pmm_manage_users`](https://github.com/cezmunsta-org/gascan-ansible/blob/b2c78bc258323497997074a3ed3c93b554d80e69/roles/pmm/defaults/main.yaml#L76) as an example

File: roles/pmm/defaults/main.yaml [hunk 2]
```diff
@@ -22,9 +22,28 @@ pmm_agent_setup_flags:
...
 
+## Alert Manager Rules Thresholds
+pmm_alertmanager_thr_PostgresqlCommitRateLow: 80
```
Comment (cezmunsta) [inline roles/pmm/defaults/main.yaml:37]: This should be defined a different way, it means that there is going to need to be a value for anything with a threshold.
I don't see why these need to be present as the values would be set when creating the alert rule and so it doesn't matter what is set in the template.

File: roles/pmm/tasks/alerting-alerts.yaml [hunk 3]
```diff
@@ -1,3 +1,11 @@
+- name: Current Alert
+  ansible.builtin.debug:
+    msg: "{{ pmm_alert_item.name }}"
+
+- name: Set Alert Filter
+  ansible.builtin.set_fact:
+    pmm_alerting_filters: "{% if pmm_alert_item.name is defined and pmm_alert_item.name in pmm_alert_filters %}{{ pmm_alert_filters[pmm_alert_item.name][\"filters\"] | default([])}}{% else %}[]{% endif %}"
```
Comment (cezmunsta) [inline roles/pmm/tasks/alerting-alerts.yaml]: What is this? This isn't how other conditional values are being set.

File: roles/pmm/vars/main.yaml [hunk 4]
```diff
@@ -27,6 +27,8 @@ pmm_grafana_alerting_notification: '{{ pmm_grafana_api }}/alertmanager/grafana/c
 
 pmm_setup_alerting: true
 notification_services: ["mysql","mongo","generic","agent","proxysql","postgresql","backup"]
+# Available alert services ["mysql","mongo","generic","agent","proxysql","postgresql","backup"]
```
Comment (cezmunsta) [inline roles/pmm/vars/main.yaml]: This requires a commented-out example. See [pmm_manage_users](https://github.com/cezmunsta-org/gascan-ansible/blob/b2c78bc258323497997074a3ed3c93b554d80e69/roles/pmm/defaults/main.yaml#L76) as an example

**Review by cezmunsta — CHANGES_REQUESTED** [review]

**Comment by cezmunsta** [comment]
https://github.com/cezmunsta-org/gascan-ansible/pull/2/files#diff-5c2b1d975c1f80e9a71539272084067ea7d6f2e1793b8d31ba0a3dad507eafa0R17-L18 seems to be missing configuration options too

File: roles/pmm/defaults/main.yaml (hunk 2 above)
Comment (theTibi) [reply roles/pmm/defaults/main.yaml:37]: No, this is set in the template not in the alert rule. In the alert rule you do not change the expression.

File: roles/pmm/tasks/alerting-alerts.yaml (hunk 3 above)
Comment (theTibi) [reply roles/pmm/tasks/alerting-alerts.yaml]: For every alert rule it has to check the filters if the alert rule name is in there, if not it need to have a default. This is the only way how I was able to achieve that but if you have better ideas please let me know.

File: roles/pmm/defaults/main.yaml (hunk 2 above)
Comment (cezmunsta) [reply roles/pmm/defaults/main.yaml:37]: So, how are rules being created from a single template with 2 different thresholds?

File: roles/pmm/tasks/alerting-alerts.yaml (hunk 3 above)
Comment (cezmunsta) [reply roles/pmm/tasks/alerting-alerts.yaml]: You seem to be missing the point, go and look how the other conditional values are being set.. certainly not like this

Comment (cezmunsta) [reply roles/pmm/tasks/alerting-alerts.yaml]: For example: 
```
roles/pmm/vars/main.yaml
77:   engine: '{{ "DISCOVER_RDS_POSTGRESQL" if group_names | intersect(["postgresql"]) else "DISCOVER_RDS_MYSQL" }}'
80:   port: '{{ 5432 if group_names | intersect(["postgresql"]) else 3306 }}'
```

File: roles/pmm/defaults/main.yaml (hunk 2 above)
Comment (theTibi) [reply roles/pmm/defaults/main.yaml:37]: that's two different template.  You can not edit the expression even on GUI when you are creating an alert.

Comment (theTibi) [reply roles/pmm/defaults/main.yaml:37]: Now I have added `params` to all the templates, thresholds can be overwritten now from inventory but the syntax is different for alert with thresholds and without them.  The API is not smart enough to use the default values from the template you have to redefine them  again when you create the alert.. anyhow now this feature is added. will push it soon.

File: roles/pmm/vars/main.yaml (hunk 4 above)
Comment (theTibi) [reply roles/pmm/vars/main.yaml]: This got removed as I reworked this part.

File: roles/pmm/vars/main.yaml [hunk 5]
```diff
@@ -24,11 +25,62 @@ pmm_grafana_alerting_rules_list: '{{ pmm_grafana_api }}/ruler/grafana/api/v1/rul
...
 pmm_grafana_alerting_notification: '{{ pmm_grafana_api }}/alertmanager/grafana/config/api/v1/alerts'
+pmm_grafana_alerting_rules_delete: '{{ pmm_grafana_alerting_rules_list }}{{ pmm_template_folder }}/'
 
 pmm_setup_alerting: true
+pmm_delete_alert_rules: false
 notification_services: ["mysql","mongo","generic","agent","proxysql","postgresql","backup"]
-# Available alert services ["mysql","mongo","generic","agent","proxysql","postgresql","backup"]
-pmm_enable_alerts_services: []
+
+pmm_alert_list: []
```
Comment (cezmunsta) [inline roles/pmm/vars/main.yaml]: Why has this moved from defaults to vars and so is no longer usable? Also, why have you made it necessary to defined as JSON????

File: roles/pmm/tasks/grafana-alerting.yaml [hunk 6]
```diff
@@ -71,10 +73,42 @@
...
+
+- name: Set pmm_existing_alert_rules
+  ansible.builtin.set_fact:
+    pmm_existing_alert_rules: '{{ pmm_get_alert_rules.json | json_query(""+ pmm_template_folder + "[*].rules[*].annotations[].rule")}}'
+
 - ansible.builtin.include_tasks: alerting-alerts.yaml
   vars:
     pmm_alert_item: '{{ item }}'
-  loop: "{{ pmm_alerting_templates['templates'] | selectattr('labels.service','in', pmm_enable_alerts_services) }}"
+  loop: "{{ pmm_alert_list | from_json | selectattr('name','defined') | list }}"
```
Comment (cezmunsta) [inline roles/pmm/tasks/grafana-alerting.yaml]: Why has this been designed to require `from_json`?

File: roles/pmm/tasks/alerting-templates.yaml [hunk 7]
```diff
@@ -19,3 +19,23 @@
...
+      "name": "{{ pmm_template_item.name }}",
+      "yaml": "{{ pmm_alerting_template_body }}"
+      }
+    force_basic_auth: yes
+    status_code:
+      - 200
+      - 409
+    body_format: json
+    validate_certs: '{{ pmm_server_insecure }}'
+  when: (pmm_template_item.name is defined and pmm_template_item.name in pmm_existing_alert_templates)
```
Comment (cezmunsta) [inline roles/pmm/tasks/alerting-templates.yaml]: Newline at EOF is missing

**Review by cezmunsta — CHANGES_REQUESTED** [review]

File: roles/pmm/defaults/main.yaml [hunk 8]
```diff
@@ -22,9 +22,79 @@ pmm_agent_setup_flags:
...
+pmm_alertmanager_thr_DiskSpaceWarning: 20
+pmm_alertmanager_thr_NodeIOUtilization: 950
+pmm_alertmanager_thr_CPUSaturationRDS: 0.8
+pmm_alertmanager_thr_StaleUpload: 93600                           # In default it is 26h (60*60*26)
+pmm_alertmanager_thr_StaleBackup: 93600                           # In default it is 26h (60*60*26)
+pmm_alertmanager_thr_StaleBackupLog: 3900                         # In default it is 65m (60*65)
+pmm_alertmanager_thr_StaleUploadLog: 3900
+
+#pmm_alert_list: {}
+pmm_alert_list:
```
Comment (cezmunsta) [inline roles/pmm/defaults/main.yaml]: @theTibi this is still JSON

**Review by cezmunsta — CHANGES_REQUESTED** [review]

**Pull request merged**